### DISCUSSION TITLE
Memory leak in Python

### DISCUSSION THREAD

**Discussion by 00031** [google/or-tools#2967] [Routing (and legacy CP) questions]
Observed memory usage increased using Routing Solver in Python 3.7.2
With `log_search = True`:

Get an assignment for the first time:
- **End search** (time = 1 ms, branches = 483, failures = 259, **memory used = 206.63 MB**, speed = 483000 branches/s)

Calling `SolveWithParameters` again to get a new assignment:
- **Start search** (**memory used = 207.35 MB**)

And the memory goes higher every time after  yielding an assignment.

**Comment by 10kc-zack** [comment]
Has there been any confirmation that this is in fact a memory leak in the Python implementation of OR-Tools? I'm hoping to use OR-Tools in a project, but won't be using Python if it has a memory leak

**Comment by shahrozecafu** [comment]
Im facing the same issue. Is there any update on this?

**Comment by ghost** [comment]
same problem, any progress on this issure?
